GhidraPython/goresym_rename.py

```diff
@@ -0,0 +1,174 @@
+#This script labels Go stripped binaries in Ghidra based on GoReSym output
+#@category Analysis
+
+from ghidra.program.model.symbol.SourceType import USER_DEFINED
+
+import json
+import re
+
+# Maps entry function to the longest _rt0_ function found by GoReSym
+CHOICE_EST_ENTRY_ADDR = "entry to _rt0_ function mapping"
+# Maps pclntab section to TabMeta VA mapping, useful only when pclntab section is present
+CHOICE_EST_PCLNTAB = "pclntab to TabMeta VA mapping"
+# Tries to map addresses against known function names in Ghidra with functions found by GoReSym
+CHOICE_EST_KNOWN_FUNCS = "known function names mapping"
+# All available choices
+CHOICES = [CHOICE_EST_ENTRY_ADDR, CHOICE_EST_PCLNTAB, CHOICE_EST_KNOWN_FUNCS]
+
+fm = currentProgram.getFunctionManager()
```

Comment:
typically i'd discourage global variables like this, but i see that ghidra sorta of encourages this by mapping magic variables into the global namespace (like `currentProgram`). *shrug*

Reply:
Since we both agree on this and you have pointed it out in your review, I have decided to put calls to `currentProgram.getFunctionManager()` into functions that use `fm` in order to remove the usage of global variables, although most ghidra scripts that I have seen use the previous approach.